File: test/lang/1-cpp/fib_2.cpp
#include "fib.h"

#include <iostream>

const unsigned short version = 2;

static long fibalgo(long value) {
	long l = 0;
	long p = 1;
	long n = value;
	for (long i = 1; i < value; i++) {
		n = l + p;
		l = p;
		p = n;
	}
	return n;
}

long fib(long value) {
	return fibalgo(value);
}

void printfib(long value) {
	std::cout << "[C++ Fibonacci Library v" << version << "] fib(" << value << ") = " << fib(value) << std::endl << std::flush;
}

File: test/lang/1-rust/fib_2.rs
use std::os::raw::c_ushort;
use std::os::raw::c_long;
use std::io::{self, Write};

#[no_mangle]
pub static version : c_ushort = 2;

fn fibalgo (value : c_long) -> c_long {
	let mut l = 0;
	let mut p = 1;
	let mut n = value;
	for _i in 1..value {
		n = l + p;
		l = p;
		p = n;
	}
	return n;
}

#[no_mangle]
pub fn fib (value : c_long) -> c_long {
	return fibalgo(value)
}

#[no_mangle]
pub fn printfib (value : c_long) {
	println!("[Rust Fibonacci Library v{}] fib({}) = {}", version, value, fibalgo(value));
	io::stdout().flush().expect("Unable to flush stdout");
}


